From 45d156fd3463924bf7d50d1656bacca160b42430 Mon Sep 17 00:00:00 2001
From: Wenchao Hao <haowenchao@huawei.com>
Date: Thu, 27 Jan 2022 09:36:39 +0800
Subject: [PATCH 1/3] Remove iscsiuio from build and install recipe

iscsiuio is to be used in conjunction with specific linux driver to
improve performance, such as QLogic NetXtreme II or QLogic CNIC driver.
It is not a necessary tool of open-iscsi.

What's more, iscsiuio used a package uio which is out of maintain now,
which would introduce CVEs unhandled. So I want to remove this tool
from open-iscsi.

This patch removes iscsiuio from build and install recipe.

Signed-off-by: Wenchao Hao <haowenchao@huawei.com>
---
 Makefile | 19 ++++---------------
 1 file changed, 4 insertions(+), 15 deletions(-)

diff --git a/Makefile b/Makefile
index 7f52cc8..5908ff9 100644
--- a/Makefile
+++ b/Makefile
@@ -17,9 +17,9 @@ rulesdir = $(etcdir)/udev/rules.d
 systemddir = $(prefix)/lib/systemd/system
 
 MANPAGES = doc/iscsid.8 doc/iscsiadm.8 doc/iscsi_discovery.8 \
-		iscsiuio/docs/iscsiuio.8 doc/iscsi_fw_login.8 doc/iscsi-iname.8 \
+		doc/iscsi_fw_login.8 doc/iscsi-iname.8 \
 		doc/iscsistart.8
-PROGRAMS = usr/iscsid usr/iscsiadm utils/iscsi-iname iscsiuio/src/unix/iscsiuio \
+PROGRAMS = usr/iscsid usr/iscsiadm utils/iscsi-iname \
 		   usr/iscsistart
 SCRIPTS = utils/iscsi_discovery utils/iscsi_fw_login utils/iscsi_offload \
 		  utils/iscsi-gen-initiatorname
@@ -29,8 +29,7 @@ IFACEFILES = etc/iface.example
 RULESFILES = utils/50-iscsi-firmware-login.rules
 SYSTEMDFILES = etc/systemd/iscsi.service \
 			   etc/systemd/iscsi-init.service \
-			   etc/systemd/iscsid.service etc/systemd/iscsid.socket \
-			   etc/systemd/iscsiuio.service etc/systemd/iscsiuio.socket
+			   etc/systemd/iscsid.service etc/systemd/iscsid.socket
 
 export DESTDIR prefix INSTALL
 
@@ -59,30 +58,22 @@ endif
 
 all: user
 
-user: iscsiuio/Makefile
+user:
 	$(MAKE) -C libopeniscsiusr
 	$(MAKE) -C utils/sysdeps
 	$(MAKE) -C utils/fwparam_ibft
 	$(MAKE) -C usr
 	$(MAKE) -C utils
-	$(MAKE) -C iscsiuio
 	@echo
 	@echo "Compilation complete                 Output file"
 	@echo "-----------------------------------  ----------------"
 	@echo "Built iSCSI daemon:                  usr/iscsid"
 	@echo "Built management application:        usr/iscsiadm"
 	@echo "Built boot tool:                     usr/iscsistart"
-	@echo "Built iscsiuio daemon:               iscsiuio/src/unix/iscsiuio"
 	@echo "Built libopeniscsiusr library:       libopeniscsiusr/libopeniscsiusr.so"
 	@echo
 	@echo "Read README file for detailed information."
 
-iscsiuio/Makefile: iscsiuio/configure iscsiuio/Makefile.in
-	cd iscsiuio; ./configure $(WITHOUT_ARG)
-
-iscsiuio/configure iscsiuio/Makefile.in: iscsiuio/configure.ac iscsiuio/Makefile.am
-	cd iscsiuio; autoreconf --install
-
 force: ;
 
 clean:
@@ -91,8 +82,6 @@ clean:
 	$(MAKE) -C utils clean
 	$(MAKE) -C usr clean
 	$(MAKE) -C libopeniscsiusr clean
-	[ ! -f iscsiuio/Makefile ] || $(MAKE) -C iscsiuio clean
-	[ ! -f iscsiuio/Makefile ] || $(MAKE) -C iscsiuio distclean
 
 # this is for safety
 # now -jXXX will still be safe
-- 
2.34.1

